core/templates/dev/head/components/top_navigation_bar/TopNavigationBarDirective.js

```diff
@@ -66,26 +67,54 @@ oppia.directive('topNavigationBar', [
           $scope.onLogoutButtonClicked = function() {
             $window.localStorage.removeItem('last_uploaded_audio_lang');
           };
-
-          $scope.profileDropdownIsActive = false;
-          $scope.onMouseoverProfilePictureOrDropdown = function(evt) {
-            angular.element(evt.currentTarget).parent().addClass('open');
-            $scope.profileDropdownIsActive = true;
-          };
-          $scope.onMouseoutProfilePictureOrDropdown = function(evt) {
-            angular.element(evt.currentTarget).parent().removeClass('open');
-            $scope.profileDropdownIsActive = false;
+          $scope.openSubmenu = function (e, menu) {
+            angular.element('.nav a').blur();
+            angular.element(e.currentTarget).focus();
+            $scope.activeMenu = menu;
           };
-          $scope.onMouseoutDropdownMenuAbout = function(evt) {
-            angular.element(evt.currentTarget)[0].blur();
+          $scope.closeSubmenu = function (evt) {
+            $scope.activeMenu = '';
+            if (evt.currentTarget.nodeName === 'UL') {
+              if (angular.element(evt.currentTarget).closest('li')
+                .find('a').is(':focus')) {
+                angular.element(evt.currentTarget).closest('li')
+                  .find('a').blur();
+              }
+            }
           };
-          $scope.onMouseoverDropdownMenu = function(evt) {
-            angular.element(evt.currentTarget).parent().addClass('open');
+          $scope.isEventOccured = function (e, targetEvent) {
```

Comment:
This function name is too vague. doesKeystrokeMatchListener or similar?

Better still, get rid of this function altogether. Replace with a declarative constant:

```
    var KEYBOARD_EVENT_TO_KEY_CODES = {
      shiftTab: {
        shiftKeyIsPressed: true,
        keyCode: 9
     },
     enter: {
       shiftKeyIsPressed: false,
       keyCode: 13
     },
     ...
  };
```
and just check membership.